// @faceless-ui/css-grid/dist/Grid/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.useGrid = void 0;
var react_1 = __importStar(require("react"));
var Cell_1 = require("../Cell");
var Settings_1 = require("../Settings");
var Context = react_1.createContext({
    cols: {
        s: 0,
        m: 0,
        l: 0,
        xl: 0,
    },
});
var useGrid = function () { return react_1.useContext(Context); };
exports.useGrid = useGrid;
var Grid = react_1.forwardRef(function (props, ref) {
    var id = props.id, children = props.children, className = props.className, style = props.style, _a = props.htmlAttributes, htmlAttributes = _a === void 0 ? {} : _a, _b = props.htmlElement, htmlElement = _b === void 0 ? 'div' : _b;
    var containingCell = Cell_1.useCell();
    var settings = Settings_1.useSettings();
    var classPrefix = settings.classPrefix;
    var value = {
        cols: {
            xl: (containingCell === null || containingCell === void 0 ? void 0 : containingCell.cols.xl) || settings.cols.xl,
            l: (containingCell === null || containingCell === void 0 ? void 0 : containingCell.cols.l) || settings.cols.l,
            m: (containingCell === null || containingCell === void 0 ? void 0 : containingCell.cols.m) || settings.cols.m,
            s: (containingCell === null || containingCell === void 0 ? void 0 : containingCell.cols.s) || settings.cols.s,
        },
    };
    var Tag = htmlElement;
    return (react_1.default.createElement(Context.Provider, { value: value },
        react_1.default.createElement(Tag, __assign({ ref: ref, id: id, className: [
                className,
                classPrefix + "__grid",
                classPrefix + "__grid--xl-cols-" + value.cols.xl,
                classPrefix + "__grid--l-cols-" + value.cols.l,
                classPrefix + "__grid--m-cols-" + value.cols.m,
                classPrefix + "__grid--s-cols-" + value.cols.s,
            ].filter(Boolean).join(' '), style: style }, htmlAttributes), children)));
});
exports.default = Grid;
