// @faceless-ui/css-grid/dist/Cell/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.useCell = void 0;
var react_1 = __importStar(require("react"));
var Grid_1 = require("../Grid");
var Settings_1 = require("../Settings");
var Context = react_1.createContext(null);
var useCell = function () { return react_1.useContext(Context); };
exports.useCell = useCell;
var Cell = react_1.forwardRef(function (props, ref) {
    var id = props.id, children = props.children, cols = props.cols, colsS = props.colsS, colsM = props.colsM, colsL = props.colsL, colsXL = props.colsXL, start = props.start, startS = props.startS, startM = props.startM, startL = props.startL, startXL = props.startXL, className = props.className, style = props.style, _a = props.htmlElement, htmlElement = _a === void 0 ? 'div' : _a, _b = props.htmlAttributes, htmlAttributes = _b === void 0 ? {} : _b;
    var colsAvailable = Grid_1.useGrid().cols;
    var _c = Settings_1.useSettings(), classPrefix = _c.classPrefix, colSettings = _c.cols;
    var colsToSpan = {
        s: colsS || colsM || colsL || cols || colsXL || colsAvailable.s,
        m: colsM || colsL || cols || colsXL || colsAvailable.m,
        l: colsL || cols || colsXL || colsAvailable.l,
        xl: cols || colsXL || colsAvailable.xl,
    };
    if (colsToSpan.l > colSettings.l)
        colsToSpan.l = colSettings.l;
    if (colsToSpan.m > colSettings.m)
        colsToSpan.m = colSettings.m;
    if (colsToSpan.s > colSettings.s)
        colsToSpan.s = colSettings.s;
    var Tag = htmlElement;
    return (react_1.default.createElement(Tag, __assign({ ref: ref, id: id, className: [
            className,
            classPrefix + "__cell",
            (start || startXL) && classPrefix + "__cell--xl-col-start-" + (start || startXL),
            startL && classPrefix + "__cell--l-col-start-" + startL,
            startM && classPrefix + "__cell--m-col-start-" + startM,
            startS && classPrefix + "__cell--s-col-start-" + startS,
            classPrefix + "__cell--xl-col-end-" + (cols || colsXL || colsAvailable.xl),
            classPrefix + "__cell--l-col-end-" + colsToSpan.l,
            classPrefix + "__cell--m-col-end-" + colsToSpan.m,
            classPrefix + "__cell--s-col-end-" + colsToSpan.s,
        ].filter(Boolean).join(' '), style: style }, htmlAttributes),
        react_1.default.createElement(Context.Provider, { value: {
                cols: colsToSpan,
            } }, children)));
});
exports.default = Cell;
